src/CharsetDetector.cs

```diff
@@ -371,43 +371,39 @@ private void FindInputState(byte[] buf, int len)
 
         private static string FindCharSetByBom(byte[] buf, int len)
         {
-            string bomSet = null;
-            if (len > 3)
+            if (len < 2 || buf.Length < 2)
+                return null;
+
+            var buf0 = buf[0];
+            var buf1 = buf[1];
+            
+            if (buf0 == 0xEF && buf1 == 0xBB && buf.Length > 2 && len > 2 && buf[2] == 0xBF)
+                return CodepageName.UTF8;
+
+            if (buf0 == 0xFE && buf1 == 0xFF)

```

Comment:
I think it's easier to read/maintain when nesting the ifs.

e.g.

`if (len > 3 && buf.Length >3){
...
   if( ....) 
   if( ....)
}
```

Reply:
Did I understand correctly?